app/Models/Foundation/Summit/SelectionPlan.php

```diff
@@ -974,4 +985,72 @@ public function setSubmissionLockDownPresentationStatusDate(?DateTime $submissio
     public function clearSubmissionLockDownPresentationStatusDate():void{
         $this->submission_lock_down_presentation_status_date = null;
     }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getPresentationActionTypes(): ArrayCollection {
+        return $this->presentation_action_types->map(function ($entity) {
+            return $entity->getType();
+        });
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getPresentationActionTypesMaxOrder(): int {
+        $criteria = Criteria::create();
+        $criteria->orderBy(['order' => 'DESC']);
+        $res = $this->presentation_action_types->matching($criteria)->first();
+        return $res === false ? 0 : $res->getOrder();
+    }
+
+    /**
+     * @param PresentationActionType $presentation_action_type
+     */
+    public function addPresentationActionType(PresentationActionType $presentation_action_type) {
+        $order = $this->getPresentationActionTypesMaxOrder();
+        $allowed_presentation_action_type = new AllowedPresentationActionType($presentation_action_type, $this, $order + 1);
+        $this->presentation_action_types->add($allowed_presentation_action_type);
+    }
+
+    /**
+     * @param PresentationActionType $presentation_action_type
+     */
+    public function removePresentationActionType(PresentationActionType $presentation_action_type) {
+        $criteria = Criteria::create();
+        $criteria->where(Criteria::expr()->eq('type', $presentation_action_type));
+        $presentation_action_type_assignment = $this->presentation_action_types->matching($criteria)->first();
+        if ($presentation_action_type_assignment === false) return;
+        $this->presentation_action_types->removeElement($presentation_action_type_assignment);
+        self::resetOrderForSelectable($this->presentation_action_types, AllowedPresentationActionType::class);
+    }
+
+    /**
+     * @param PresentationActionType $presentation_action_type
+     * @param int $order
+     * @throws ValidationException
+     */
+    public function updateOrder(PresentationActionType $presentation_action_type, int $order) {
```

Comment:
take a look on the same class here
https://github.com/OpenStackweb/summit-api/blob/417557f69bc7a3751993f39a84a44a02cb534750/app/Models/Foundation/Summit/SelectionPlan.php#L654
please lets keeps the naming convention of the method

Reply:
done